### PR TITLE
feat(docs): Add Icon viewer page

### PR DESCRIPTION
**Pull request by mnajdova**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [ ] Include a change request file using `$ yarn change`

#### Description of changes

(give an overview)

#### Focus areas to test

(optional)

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/fluentui/pull/12645)

### CODE REVIEW

File: packages/fluentui/docs/src/components/Sidebar/Sidebar.tsx [hunk 1]
```diff
@@ -165,6 +165,15 @@ class Sidebar extends React.Component<any, any> {
               to: '/shorthand-props',
             },
           },
+          {
```
Comment (mnajdova) [inline packages/fluentui/docs/src/components/Sidebar/Sidebar.tsx:168]: Now that we have the viewer we may even hide from public the example with all icons, and add link on the SvgIcon's docs page for the icons viewer... I am open for other ideas/suggestions here

File: packages/fluentui/docs/src/views/IconViewer.tsx [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+        </Flex.Item>
+      </Flex>
+      <Flex column gap="gap.medium" padding="padding.medium">
+        <Header content="Looking for particular icon?" style={{ textAlign: 'center' }} />
+        <Input
+          inverted
+          fluid
+          clearable
+          icon={<SearchIcon />}
+          placeholder={`Search ${icons.length} icons`}
```
Comment (layershifter) [inline packages/fluentui/docs/src/views/IconViewer.tsx:183]:
```suggestion
          placeholder={`Search across ${icons.length} icons`}
```

?

File: packages/fluentui/docs/src/views/IconViewer.tsx [hunk 3]
```diff
@@ -0,0 +1,239 @@
...
+              >
+                <Card.Header>
+                  <Flex gap="gap.small" column hAlign="center">
+                    <Icon size="largest" />
+                    <br />
+                    <CopyToClipboard value={`<${Icon.displayName} />`} timeout={3000}>
+                      {(active, onClick) => (
+                        <Box
+                          as="code"
+                          onClick={onClick}
```
Comment (layershifter) [inline packages/fluentui/docs/src/views/IconViewer.tsx:205]:
```suggestion
                          onClick={onClick}
                          title='Click to copy JSX to your clipboard'
```

File: packages/fluentui/docs/src/views/IconViewer.tsx [hunk 4]
```diff
@@ -0,0 +1,239 @@
...
+          icon={<SearchIcon />}
+          placeholder={`Search ${icons.length} icons`}
+          iconPosition="end"
+          role="search"
+          onChange={handleQueryChange}
+          value={query}
+        />
+        <Flex wrap gap="gap.medium" styles={iconFlexStyles}>
+          {filteredIcons.map(Icon => (
+            <Flex.Item align="center" key={Icon.displayName}>
```
Comment (layershifter) [inline packages/fluentui/docs/src/views/IconViewer.tsx:191]: Should we extract this to a component?

File: packages/fluentui/docs/src/views/IconViewer.tsx [hunk 5]
```diff
@@ -0,0 +1,239 @@
...
+    acc.push(exports[exportName]);
+  }
+
+  return acc;
+}, []);
+
+const exampleCode = `
+import { QnaIcon } from '@fluentui/react-icons-northstar';
+
+const Icons = () => (
```
Comment (layershifter) [inline packages/fluentui/docs/src/views/IconViewer.tsx:38]:
```suggestion
const Example = () => (
```

File: packages/fluentui/docs/src/views/IconViewer.tsx [hunk 6]
```diff
@@ -0,0 +1,239 @@
...
+        <CodeSnippet mode="jsx" value={exampleCode} />
+      </Card.Body>
+    </Card>
+  );
+};
+
+const PlaygroundCard = props => {
+  const { QnaIcon, EditIcon } = exports;
+
+  const [bordered, setBordered] = React.useState(false);
```
Comment (layershifter) [inline packages/fluentui/docs/src/views/IconViewer.tsx:77]: It makes sense to use a single `useReducer` there instead...

File: packages/fluentui/docs/src/views/IconViewer.tsx [hunk 7]
```diff
@@ -0,0 +1,239 @@
...
+              <Dropdown
+                items={['smallest', 'smaller', 'small', 'medium', 'large', 'larger', 'largest']}
+                value={size}
+                onChange={(e, d) => setSize(d.value.toString())}
+              />
+            </div>
+            <Divider />
+            <div>
+              <Text>
+                Checkout our <Link to="components/svg-icon/definition">icon</Link> docs for more examples!
```
Comment (layershifter) [inline packages/fluentui/docs/src/views/IconViewer.tsx:127]: Let's add some styling there, it's not clear that it's a link :(

image

File: packages/fluentui/docs/src/views/IconViewer.tsx [hunk 8]
```diff
@@ -0,0 +1,239 @@
...
+            <Flex hAlign="center" vAlign="center">
+              <Flex.Item>
+                <div>
+                  <QnaIcon
+                    size={size as SvgIconSizeValue}
+                    outline={outline}
+                    bordered={bordered}
+                    circular={circular}
+                    rotate={rotate}
+                  />
```
Comment (layershifter) [inline packages/fluentui/docs/src/views/IconViewer.tsx:141]: We can extract this to an `element` variables and then reuse `ComponentPlaygroundSnippet`:

image

```jsx
<ComponentPlaygroundSnippet element={element} />
```

**Pull request merged**